src/multiplexed-session.ts

```diff
@@ -0,0 +1,241 @@
+          isInstanceNotFoundError(err) ||
+          isCreateSessionPermissionError(err) ||
+          isDefaultCredentialsNotSetError(err) ||
+          isProjectIdNotSetInEnvironmentError(err)
+        ) {
+          return;
+        }
+        this.emit('error', err);
+      });
+  }
+
+  /**
+   * Creates a new multiplexed session.
+   *
+   * This method sends a request to the database to create a new session with multiplexing enabled.
+   * The response from the database would be an array, the first value of the array will be containing the multiplexed session.
+   *
+   * @returns {Promise<void>} A Promise that resolves when the session has been successfully created and assigned, an event
+   * `mux-session-available` will be emitted to signal that the session is ready.
+   *
+   * In case of error, an error will get emitted along with the erorr event.
+   *
+   * @private
+   */
+  async _createSession(): Promise<void> {
+    try {
+      const [createSessionResponse] = await this.database.createSession({
+        multiplexed: true,
+      });
+      this._multiplexedSession = createSessionResponse;
+      this.emit('mux-session-available');
+    } catch (e) {
+      this.emit('error', e);
+    }
+  }
+
+  /**
+   * Maintains the multiplexed session by periodically refreshing it.
+   *
+   * This method sets up a periodic refresh interval for maintaining the session. The interval duration
+   * is determined by the @param refreshRate option, which is provided in days.
+   * The default value is 7 days.
+   *
+   * @returns {void} This method does not return any value.
+   *
+   * @throws {Error} If the `refreshRate` option is not defined, this method could throw an error.
+   */
+  _maintain(): void {
+    const refreshRate = this.refreshRate! * 24 * 60 * 60000;
+    this._refreshHandle = setInterval(() => this._refresh(), refreshRate);
+    this._refreshHandle.unref();
+  }
+
+  /**
+   * Refreshes the session by checking its expiration time and recreating the session if expired.
+   * Every 7th day a new Multiplexed Session will get created. However, a Multiplexed Session will
+   * be alive for 30 days in the backend.
+   *
+   * @returns {Promise<void>} A Promise that resolves once the refresh process is completed.
+   *                          If the session is expired, a new session will be created.
+   *
+   * @throws {Error} If there is an issue with retrieving the session metadata or calculating the expiration time.
+   */
+  async _refresh(): Promise<void> {
+    const metadata = await this._multiplexedSession?.getMetadata();
+    const createTime =
+      parseInt(metadata![0].createTime.seconds) * 1000 +
+      metadata![0].createTime.nanos / 1000000;
+
+    // Calculate expiration time (7 days after session creation)
+    const expireTime = createTime + this.refreshRate * 24 * 60 * 60 * 1000;
+
+    // If the current time exceeds the expiration time, create a new session
+    if (Date.now() > expireTime) {
```

Comment:
Yeah, we don't need this check here

Reply:
We don't even need this _refresh method since, we are sure that the interval is 7 days after the mux session creation, I have modified the _maintain method to call directly _createSession in a interval of 7 days